shared/data/fingerprint-protection.js

```diff
@@ -283,6 +283,17 @@
             let e = document.createElement('script')
             e.textContent = scriptToInject
             const elem = document.head || document.documentElement
+            try {
+                const contentType = elem.ownerDocument.contentType
+                if (contentType === 'application/xml' ||
+                    contentType === 'application/json' ||
+                    contentType === 'text/xml' ||
+                    contentType === 'text/json') {
+                    return
+                }
+            } catch (e) {
+                // if we can't find content type, go ahead with injection.
+            }
```

Comment:
instead of doing this check here, I think it'd be better to do it couple of lines below and avoid creating a window resize listener and the injection script in the first place e.g.

```js
if (!isXMLorJSON()) {
    window.addEventListener('resize', function () {
        const windowScript = setWindowDimensions()
        inject(windowScript, true)
    })

    const injectionScript = buildInjectionScript()
    inject(injectionScript)
}
```

or do it at the very beginning of the script and just `return` early like Dave did below.

Reply:
Good thinking, made the changes.